### ISSUE TITLE
Add to FAQ: how to use this plugin in a Typescript project?

### ISSUE THREAD

**Issue by checkmatez** [mysticatea/eslint-plugin-node#205]
I am probably missing something, since I can't make it work. Linter should lint source files and not output, right?

Running Eslint prints out errors: "Parsing error: ImportDeclaration should appear when the mode is ES6 and in the module context" and "Import and export declarations are not supported yet".
I don't have "type" in package.json, and my eslintrc contains:
```js
{
parser: '@typescript-eslint/parser',
parserOptions: {
    ecmaVersion: 2018,
    project: 'tsconfig.json',
  }
// ... plugins and extends
}
```

**Comment by mysticatea** [comment]
Thank you for your report.

But it looks like non-related issue to this plugin.
ESLint needs `parserOptions.sourceType` setting to handle ES modules.

**Comment by checkmatez** [comment]
I tried with
```js
{
  parser: '@typescript-eslint/parser',
  parserOptions: {
    ecmaVersion: 2018,
    project: 'tsconfig.json',
    sourceType: 'module',
  }
}
```
and still received errors on `node/no-missing-import` and `node/no-unsupported-features/es-syntax` rules.
I will close the issue, but I am still looking for a proper way to use eslint-plugin-node with Typescript.

**Comment by MichielDeMey** [comment]
Be sure to also configure the `.ts` extension in your `.eslintrc.js`.
```js
settings: {
    node: {
        "resolvePaths": [__dirname],
        "tryExtensions": [".js", ".json", ".node", ".ts"]
    }
  }
```

**Comment by checkmatez** [comment]
Thank you, @MichielDeMey ! That is what I was missing.

I ended up with config close to:
```js
module.exports = {
  parser: '@typescript-eslint/parser',
  parserOptions: {
    ecmaVersion: 2018,
    project: 'tsconfig.json',
  },
  plugins: ['@typescript-eslint',  'node'],
  extends: [
    'eslint:recommended',
    'plugin:@typescript-eslint/eslint-recommended',
    'plugin:@typescript-eslint/recommended',
    'plugin:node/recommended'
  ],
  settings: {
    node: {
      tryExtensions: ['.js', '.json', '.node', '.ts'],
    },
  },
  rules: {
    'node/no-unsupported-features/es-syntax': ['error', { ignores: ['modules'] }],
  }
}
```
